modules/pool/lib/apr-data-sources/morpho-rewards-apr.service.ts

```diff
@@ -0,0 +1,83 @@
+import { PoolAprService } from '../../pool-types';
+import { PrismaPoolWithTokens } from '../../../../prisma/prisma-types';
+import { prisma } from '../../../../prisma/prisma-client';
+import { PrismaPoolAprItemGroup, PrismaPoolAprType } from '@prisma/client';
+import { morphoApiClient } from './morpho-api-client';
+import murmurhash from 'murmurhash';
+
+const MORPHO_TOKEN = '0x58d97b57bb95320f9a05dc918aef65434969c2b2';
+
+export class MorphoRewardsAprService implements PoolAprService {
+    public getAprServiceName(): string {
+        return 'MorphoRewardsAprService';
+    }
+
+    public async updateAprForPools(pools: PrismaPoolWithTokens[]): Promise<void> {
+        const aprItems = await this.getAprItems(pools);
+
+        await prisma.$transaction(
+            aprItems.map((item) =>
+                prisma.prismaPoolAprItem.upsert({
+                    where: { id_chain: { id: item.id, chain: item.chain } },
+                    update: {
+                        apr: item.apr,
+                    },
+                    create: item,
+                }),
+            ),
+        );
+    }
+
+    private async getAprItems(pools: PrismaPoolWithTokens[]) {
+        // Get Morpho rewards
+        const morphoRewards = await morphoApiClient.rewardTokens();
+        const morphoApr = await morphoApiClient.morphoApr();
+
+        // Find all pools with Morpho vault tokens
+        const morphoAddresses = Object.keys(morphoRewards);
```

Comment:
minor: can we call this `morphoVaultAddresses`? was confused here

Reply:
changed